tutorials/Tutorial_NumericalModel_3D.jl

```diff
@@ -0,0 +1,96 @@
+        Origin=(-0,0,0),
+        T=SpreadingRateTemp(SpreadingVel=3, MORside="right"), StrikeAngle=30);
+
+# And an an inclined part:
+AddBox!(Phases, Temp, Grid; xlim=(0,300), ylim=(-400, 400.0), zlim=(-80.0, 0.0), phase = lith, 
+        Origin=(-0,0,0),
+        T=McKenzie_subducting_slab(Tsurface=0,v_cm_yr=3), DipAngle=30, StrikeAngle=30);
+
+# Add them to the `CartData` dataset:
+Grid = addField(Grid,(;Phases, Temp))
+
+# Which looks like
+Write_Paraview(Grid,"Grid3D_FreeSubduction");
+# ![Mechanical3D_Tutorial_1](../assets/img/Mechanical3D_Tutorial_1.png) 
+
+
+# #### More sophisticated setup
+
+# Next, lets consider a somewhat more complicated setup 
+Phases = fill(2,nx,ny,nz);
+Temp = fill(1350.0, nx,ny,nz);
+
+# Overriding plate
+lith_cont = LithosphericPhases(Layers=[30 200 10], Phases=[3 4 2], Tlab=1250)
+AddBox!(Phases, Temp, Grid; xlim=(200,1000), ylim=(-1000, 0.0), zlim=(-240.0, 0.0), phase = lith_cont, T=HalfspaceCoolingTemp(Age=150));
+
+lith_cont = LithosphericPhases(Layers=[30 200 10], Phases=[5 6 2], Tlab=1250)
+AddBox!(Phases, Temp, Grid; xlim=(200,1000), ylim=(0, 1000),    zlim=(-240.0, 0.0), phase = lith_cont, T=HalfspaceCoolingTemp(Age=200));
+
+# Define a ridge with transform faults
+lith = LithosphericPhases(Layers=[15 45 10], Phases=[0 1 2], Tlab=1250)
+AddBox!(Phases, Temp, Grid; xlim=(-800 , 200), ylim=(-1000, -400.0), zlim=(-80.0, 0.0), phase = lith, T=SpreadingRateTemp(SpreadingVel=3));
+AddBox!(Phases, Temp, Grid; xlim=(-1000,-800), ylim=(-1000, -400.0), zlim=(-80.0, 0.0), phase = lith, T=SpreadingRateTemp(SpreadingVel=3,MORside="right"));
+
+AddBox!(Phases, Temp, Grid; xlim=(-700,  200), ylim=(-400, 200.0), zlim=(-80.0, 0.0), phase = lith, T=SpreadingRateTemp(SpreadingVel=3));
+AddBox!(Phases, Temp, Grid; xlim=(-1000,-700), ylim=(-400, 200.0), zlim=(-80.0, 0.0), phase = lith, T=SpreadingRateTemp(SpreadingVel=3,MORside="right"));
+
+AddBox!(Phases, Temp, Grid; xlim=(-650,  200), ylim=(200, 1000.0), zlim=(-80.0, 0.0), phase = lith, T=SpreadingRateTemp(SpreadingVel=3));
+AddBox!(Phases, Temp, Grid; xlim=(-1000,-650), ylim=(200, 1000.0), zlim=(-80.0, 0.0), phase = lith, T=SpreadingRateTemp(SpreadingVel=3,MORside="right"));
+
+# Inclined part of slab
+lith = LithosphericPhases(Layers=[15 15 45 40], Phases=[7 0 1 2], Tlab=1250)
+
+AddBox!(Phases, Temp, Grid; xlim=(200,700), ylim=(-1000, 0.0), zlim=(-200.0, 15), phase = lith,  T=ConstantTemp(T=1350), DipAngle=30);
+AddBox!(Phases, Temp, Grid; xlim=(200,700), ylim=(0, 100.0), zlim=(-200.0, 15), phase = lith,  T=ConstantTemp(T=1350), DipAngle=35);
+
+AddBox!(Phases, Temp, Grid; xlim=(200,700), ylim=(-1000, 0.0), zlim=(-110.0, 15), phase = lith,  T=McKenzie_subducting_slab(Tsurface=0,v_cm_yr=3), DipAngle=30);
+AddBox!(Phases, Temp, Grid; xlim=(200,700), ylim=(0, 1000), zlim=(-110.0, 15), phase = lith,  T=McKenzie_subducting_slab(Tsurface=0,v_cm_yr=3), DipAngle=35);
+
+
+ind=findall(isnan.(Temp)); Temp[ind] .= 0;
```

Comment:
```suggestion
map!(x -> isnan(x) ? zero(x) : x, Temp, Temp);
```
This is very MATLAB style (and allocates few intermediate arrays), could be replace by something more Julian.

Reply:
hmm, I personally find your Julia way pretty hard to understand...
Can this be done in a less cryptic way?

Reply:
What about?
```julia
replace!(Temp, NaN => 0.0)
```

Reply:
ok that is indeed easier. changed that accordingly